papers/Vvmax/calc_vvmax.py

```diff
@@ -0,0 +1,764 @@
+    plt.savefig("frb_b_values.pdf")
+    plt.close()
+
+def airy(x):
+    """
+    airy disk function
+    """
+    
+    val = (2. * BessJ(1,x)/x)**2
+    return val
+  
+def load_beamshape():
+    """
+    loads askap beamshape info
+    """
+    import os
+    sdir = os.path.join(resource_filename('zdm', 'data'), 'BeamData/')
+    bvals = np.load(sdir+"lat50_log_bins.npy")
+    omegab = np.load(sdir + "lat50_log_hist.npy")
+    return bvals,omegab
+
+def init_cos():
+    """
+    
+    """
+    state = loading.set_state()
+    state.cosmo.H0 = 70.0
+    state.cosmo.Omega_b = 0.0486
+    cos.set_cosmology(state)
+    cos.init_dist_measures()
+    return state
+    
+def get_macquart_z(DMeg,state,DMhost):
+    """
+    gets z(DM) from the Macquart relation
+    """
+    
+    # gets z from macquart relation
+    zvals=np.linspace(1e-3,2,2000)
+    
+    macquart_relation=pcosmic.get_mean_DM(zvals, state)
+    hosts = DMhost/(1+zvals)
+    macquart_relation += hosts #adding the host contribution as a function of z 
+    splines = interpolate.CubicSpline(macquart_relation,zvals)
+    zFRBs = splines(DMeg)
+    #print(zFRBs)
+    
+    return zFRBs
+
+
+def get_DM_cosmic(z,state):
+    """
+    gets z(DM) from the Macquart relation
+    """
+    
+    # gets z from macquart relation
+    zvals=np.linspace(1e-3,3,3000)
+    
+    macquart_relation=pcosmic.get_mean_DM(zvals, state)
+    splines = interpolate.CubicSpline(zvals,macquart_relation)
+    DMcosmic = splines(z)
+    
+    return DMcosmic
+
+
+def plot_z_error(myz,waynez):    
+    """
+    Plots error in z from excluding f_d
+    """
+    
+    plt.figure()
+    plt.scatter(waynez,myz)
+    plt.xlabel("Wayne")
+    plt.ylabel("Me")
+    plt.xlim(0.,1.5)
+    plt.ylim(0.,1.5)
+    plt.savefig("redshift_error.pdf")
+    plt.close()
+    
+    
+    
+def load_loc_frb_data(fname):
+    """
+    Loads table of localised FRBs
+    """
+    data = np.loadtxt(fname)
+    names = data[:,0]
+    DM = data[:,2]
+    freqMHz = data[:,4]
+    SNR = data[:,1]
+    #SNRth = data[:,4]
+    DMG = data[:,3]
+    tsamp = data[:,5]
+    width = data[:,6]
+    BeamThetaDeg = data[:,7]
+    oldFnu = data[:,8]
+    zloc = data[:,9]
+    Nants = data[:,10]
+    newFnu = data[:,11]
+    
+    SNRth = np.full([newFnu.size],9.5)
+    
+    return names,DM,freqMHz,SNR,SNRth,DMG,tsamp,width,BeamThetaDeg,newFnu,zloc,Nants
+    
+
+def load_unloc_frb_data(filename):
+    """
+    Loads table of unlocalised FRBs
+    
+    Uses new fluence data from Ryan
+    """
+    data = np.loadtxt(filename)
+    names = data[:,0]
+    DM = data[:,1]
+    freqMHz = data[:,2]
+    SNR = data[:,3]
+    SNRth = data[:,4]
+    DMG = data[:,5]
+    tsamp = data[:,6]
+    width = data[:,7]
+    BeamThetaDeg = data[:,8]
+    oldFnu = data[:,9]
+    zDM = data[:,10]
+    Nants = data[:,11]
+    newFnu = data[:,12]
+    return names,DM,freqMHz,SNR,SNRth,DMG,tsamp,width,BeamThetaDeg,newFnu,zDM,Nants
+
+
+
+global macquartz
```

Comment:
a `global` here, a `global` there, makes code hard to follow and debug..

Reply:
No globals in the part of the code in the main repo!